hw/ip/kmac/doc/_index.md

```diff
@@ -256,7 +256,97 @@ The refresh does not block the internal entropy expansion operation.
 
 ### Error Report
 
-_TBD_
+This section explains the errors KMAC HWIP raises during the hasing operations, their meanings, and the error handling process.
+
+KMAC HWIP has the error checkers in its internal datapath.
+If the checkers detect errors, whether they are triggered by the SW mis-configure, or HW malfunctions, they report the error to {{< regref "ERR_CODE" >}} and raises an `kmac_error` interrupt.
+Each error code gives debugging information at the lower 24 bits of {{< regref "ERR_CODE" >}}.
+
+Value | Error Code | Description
+------|------------|-------------
+0x01  | KeyNotValid | In KMAC mode with the sideloaded key, the IP raises an error if the sideloaded secret key is not ready.
+0x02  | SwPushedMsgFifo | MsgFifo is updated not in the Message Feed state.
+0x03  | SwIssuedCmdInAppActive | SW issued a command while the application interface is being used
+0x04  | WaitTimerExpired | EDN has not responded within the wait timer limit.
+0x05  | IncorrectEntropyMode | When SW sets `entropy_ready`, the `entropy_mode` is neither SW nor EDN.
+0x06  | UnexpectedModeStrength | SHA3 mode and Keccak Strength combination is not expected.
+0x07  | IncorrectFunctionName | In KMAC mode, the PREFIX has the value other than `encoded_string("KMAC")`
+0x08  | SwCmdSequence | Sw does not follow the guided sequence, `start` -> `process` -> {`run` ->} `done`
+0x80  | Sha3Control | SW may receive Sha3Control error along with `SwCmdSequence` error. Can be ignored.
+
+#### KeyNotValid (0x01)
+
+The `KeyNotValid` error is raised in the application interface module.
+When a KMAC application requests a hashing operation, the module checks if the sideloaded key is ready.
+If the key is not ready, the module reports `KeyNotValid` error and moves to dead-end state and waits the IP reset.
+
+This eror does not provide any additional information.
+
+#### SwPushedMsgFifo (0x02)
+
+The `SwPushedMsgFifo` error happens when the Message FIFO recevies TL-UL transactions while the application interface is in active.
```

Comment:
the last parts of this sentence may be misleading as it could be interpreted as "inactive" (and I believe this would be wrong, right?).

how about "while the application interface is busy" or "in an active state"?